src/main/java/ee/carlrobert/llm/client/watsonx/WatsonxAuthenticator.java

```diff
@@ -0,0 +1,128 @@
+package ee.carlrobert.llm.client.watsonx;
+
+import static ee.carlrobert.llm.client.DeserializationUtil.OBJECT_MAPPER;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.Base64;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+public class WatsonxAuthenticator {
+
+  IBMAuthBearerToken bearerToken;
+  OkHttpClient client;
+  Request request;
+  Request expiryRequest;
+  Boolean isZenApiKey = false;
+
+  // Watsonx SaaS
+  public WatsonxAuthenticator(String apiKey) {
+    this.client = new OkHttpClient().newBuilder()
+        .build();
+    MediaType mediaType = MediaType.parse("application/x-www-form-urlencoded");
+    RequestBody body = RequestBody.create(mediaType,
+        "grant_type=urn:ibm:params:oauth:grant-type:apikey&apikey=" + apiKey);
+    this.request = new Request.Builder()
+        .url("https://iam.cloud.ibm.com/identity/token")
+        .method("POST", body)
+        .addHeader("Content-Type", "application/x-www-form-urlencoded")
+        .build();
+    try {
+      Response response = client.newCall(request).execute();
+      this.bearerToken = OBJECT_MAPPER.readValue(response.body().string(),
+          IBMAuthBearerToken.class);
+    } catch (IOException e) {
+      System.out.println(e);
+    }
+  }
+
+  // Zen API Key
+  public WatsonxAuthenticator(String username, String zenApiKey) {
+    IBMAuthBearerToken token = new IBMAuthBearerToken();
+    String tokenStr = Base64.getEncoder().encode((username + ":" + zenApiKey).getBytes())
+        .toString();
+    token.setAccessToken(tokenStr);
+    this.bearerToken = token;
+    this.isZenApiKey = true;
+  }
+
+  // Watsonx API Key
+  public WatsonxAuthenticator(String username, String apiKey,
```

Comment:
The constructor is too complex. Can we break it down into smaller units for better readability?